### PR TITLE
JIT: revise how synthesis treats handler regions

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Instead of giving hander regions a fraction of the entry weight, give them a small fixed weight. This is intended to combat the lack of profile propagation out of handler regions, where there are currently sometimes weight discontinuities large enough to cause profile check asserts.

Contributes to #93020.

A small number of [Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=638988&view=ms.vss-build-web.run-extensions-tab).

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@amanasifkhalid PTAL
cc @dotnet/jit-contrib 

This will be needed to land #100869

File: src/coreclr/jit/fgprofilesynthesis.h [hunk 1]
```diff
@@ -45,7 +45,7 @@ class ProfileSynthesis
     {
     }
 
-    static constexpr weight_t exceptionScale        = 0.001;
+    static constexpr weight_t exceptionWeight       = 0.00001;
```
Comment (amanasifkhalid) [inline src/coreclr/jit/fgprofilesynthesis.h:48]: I'm a bit surprised to see this fixed weight is smaller than the scale factor it's replacing. Did you get this number through trial and error, or?

**Review by amanasifkhalid — APPROVED** [review]
LGTM, thanks!

File: src/coreclr/jit/fgprofilesynthesis.h (hunk 1 above)
Comment (AndyAyersMS) [reply src/coreclr/jit/fgprofilesynthesis.h:48]: Nothing magic here. Just wanted a value that is small enough to ignore in an absolute sense.

Comment (amanasifkhalid) [reply src/coreclr/jit/fgprofilesynthesis.h:48]: I see -- so close to zero, but it won't mark the block with `BBF_RUN_RARELY`.

**Pull request merged**